### PR TITLE
worktree: Remove and Move methods

### PR DESCRIPTION
**Pull request by mcuadros**
Requires https://github.com/src-d/go-billy/pull/29

### CODE REVIEW

File: worktree_status.go [hunk 1]
```diff
@@ -72,6 +74,8 @@ func (w *Worktree) status(commit plumbing.Hash) (Status, error) {
 			fs.Staging = Unmodified
 		}
 
+		fmt.Println(ch, a)
```
Comment (alcortesm) [inline worktree_status.go]: leftover?

File: worktree_status.go [hunk 2]
```diff
@@ -266,10 +271,61 @@ func (w *Worktree) doUpdateFileToIndex(e *index.Entry, filename string, h plumbi
...
+
 	if err != nil {
 		return err
 	}
 
 	fillSystemInfo(e, info.Sys())
 	return nil
 }
+
+func (w *Worktree) Remove(path string) (plumbing.Hash, error) {
```
Comment (alcortesm) [inline worktree_status.go]: Document this method.

File: worktree_status.go [hunk 3]
```diff
@@ -266,10 +271,61 @@ func (w *Worktree) doUpdateFileToIndex(e *index.Entry, filename string, h plumbi
...
+func (w *Worktree) deleteFromFilesystem(path string) error {
+	err := w.fs.Remove(path)
+	if os.IsNotExist(err) {
+		return nil
+	}
+
+	return err
+}
+
+func (w *Worktree) Move(from, to string) (plumbing.Hash, error) {
```
Comment (alcortesm) [inline worktree_status.go]: Document this method.

File: plumbing/format/index/index_test.go [hunk 4]
```diff
@@ -20,3 +20,20 @@ func (s *IndexSuite) TestIndexEntry(c *C) {
 	c.Assert(e, IsNil)
 	c.Assert(err, Equals, ErrEntryNotFound)
 }
+
+func (s *IndexSuite) TestIndexRemove(c *C) {
```
Comment (ajnavarro) [inline plumbing/format/index/index_test.go]: TestIndexRemoveNonexistentEntry?

File: worktree_test.go [hunk 5]
```diff
@@ -590,3 +590,68 @@ func (s *WorktreeSuite) TestAddUnmodified(c *C) {
 	c.Assert(hash.String(), Equals, "c192bd6a24ea1ab01d78686e417c8bdc7c3d197f")
 	c.Assert(err, IsNil)
 }
+
+func (s *WorktreeSuite) TestRemove(c *C) {
```
Comment (ajnavarro) [inline worktree_test.go]: TestRemoveNonexistentElement?

File: worktree_test.go [hunk 6]
```diff
@@ -590,3 +590,68 @@ func (s *WorktreeSuite) TestAddUnmodified(c *C) {
...
+	c.Assert(hash.String(), Equals, "c192bd6a24ea1ab01d78686e417c8bdc7c3d197f")
+	c.Assert(err, IsNil)
+
+	status, err := w.Status()
+	c.Assert(err, IsNil)
+	c.Assert(status, HasLen, 1)
+	c.Assert(status.File("LICENSE").Staging, Equals, Deleted)
+}
+
+func (s *WorktreeSuite) TestRemoveDeletedFromWorktree(c *C) {
```
Comment (ajnavarro) [inline worktree_test.go]: TestRemoveNonexistentDeletedElement?

File: plumbing/format/index/index_test.go (hunk 4 above)
Comment (mcuadros) [inline plumbing/format/index/index_test.go]: this is the only test for Remove so I rather keep like this

File: worktree_test.go (hunk 5 above)
Comment (mcuadros) [inline worktree_test.go]: this is the only test for Remove so I rather keep like this

File: plumbing/format/index/index_test.go (hunk 4 above)
Comment (mcuadros) [inline plumbing/format/index/index_test.go]: This is already tested

Comment (ajnavarro) [inline plumbing/format/index/index_test.go]: :+1: I didn't know, because the codecov comment is not appearing...

**Pull request merged**